### PR TITLE
#260 Added system-wide loggingLevel configuration for Agents and Masters

### PR DESCRIPTION
**Pull request by frankscholten**

### CODE REVIEW

File: minimesos/src/main/groovy/com/containersol/minimesos/config/ClusterConfig.groovy [hunk 1]
```diff
@@ -53,4 +53,14 @@ class ClusterConfig extends GroovyBlock {
         delegateTo(marathon, cl)
     }
 
+    void setLoggingLevel(String loggingLevel) {
+         if (!loggingLevel.equals("WARNING") && !loggingLevel.equals("INFO") && !loggingLevel.equals("ERROR")) {
+             throw new RuntimeException("Property 'loggingLevel' can only have the values INFO, WARNING or ERROR")
```
Comment (sadovnikov) [inline minimesos/src/main/groovy/com/containersol/minimesos/config/ClusterConfig.groovy]: * WARN is normally used for warnings. I think we should allow both and I would include ERR too
* I would make it case insensitive

File: minimesos/src/main/java/com/containersol/minimesos/main/CommandUp.java [hunk 2]
```diff
@@ -209,6 +209,7 @@ private void updateWithParameters(ClusterConfig clusterConfig) {
         // Mesos Master
         MesosMasterConfig masterConfig = (clusterConfig.getMaster() != null) ? clusterConfig.getMaster() : new MesosMasterConfig();
         masterConfig.setImageTag(getMesosImageTag());
+        masterConfig.setLoggingLevel(clusterConfig.getLoggingLevel());
```
Comment (sadovnikov) [inline minimesos/src/main/java/com/containersol/minimesos/main/CommandUp.java]: I believe this is wrong: with changes in `MesosContainerConfig` users can specify logging level on container level. However this code will overwrite it with default or specified level from cluster. This one should work like "number of agents" - if was not specified on container level then take from cluster.

File: minimesos/src/main/java/com/containersol/minimesos/main/CommandUp.java [hunk 3]
```diff
@@ -222,6 +223,7 @@ private void updateWithParameters(ClusterConfig clusterConfig) {
         for (int i = 0; i < getNumAgents(); i++) {
             MesosAgentConfig agentConfig = (agentConfigs.size() > i) ? agentConfigs.get(i) : new MesosAgentConfig();
             agentConfig.setImageTag(getMesosImageTag());
+            agentConfig.setLoggingLevel(clusterConfig.getLoggingLevel());
```
Comment (sadovnikov) [inline minimesos/src/main/java/com/containersol/minimesos/main/CommandUp.java]: I believe this is wrong: with changes in MesosContainerConfig users can specify logging level on container level. However this code will overwrite it with default or specified level from cluster. This one should work like "number of agents" - if was not specified on container level then take from cluster.

File: minimesos/src/main/java/com/containersol/minimesos/mesos/MesosMaster.java [hunk 4]
```diff
@@ -48,6 +48,7 @@ public int getPortNumber() {
         TreeMap<String,String> envs = new TreeMap<>();
         envs.put("MESOS_QUORUM", "1");
         envs.put("MESOS_ZK", getFormattedZKAddress());
+        envs.put("MESOS_LOGGING_LEVEL", config.getLoggingLevel());
```
Comment (sadovnikov) [inline minimesos/src/main/java/com/containersol/minimesos/mesos/MesosMaster.java]: Why do we play with environment variables in both `MesosMaster` and `MesosContainer`?

File: minimesos/src/test/groovy/com/containersol/minimesos/config/ConfigParserTest.groovy [hunk 5]
```diff
@@ -17,6 +17,20 @@ public class ConfigParserTest {
...
+
+            loggingLevel = "WARNING"
+
+        }
+        """
+
+        assertEquals("WARNING", parser.parse(config).getLoggingLevel())
+    }
+
+    @Test
```
Comment (sadovnikov) [inline minimesos/src/test/groovy/com/containersol/minimesos/config/ConfigParserTest.groovy]: Please, add test to ensure possibility to change logging level for specific mesos container.

File: minimesos/src/main/groovy/com/containersol/minimesos/config/ClusterConfig.groovy (hunk 1 above)
Comment (frankscholten) [inline minimesos/src/main/groovy/com/containersol/minimesos/config/ClusterConfig.groovy]: WARNING, INFO and ERROR are the supported logging levels for Mesos http://mesos.apache.org/documentation/latest/configuration

Case insensitive, good point

File: minimesos/src/main/java/com/containersol/minimesos/main/CommandUp.java (hunk 2 above)
Comment (frankscholten) [inline minimesos/src/main/java/com/containersol/minimesos/main/CommandUp.java]: Ok

File: minimesos/src/main/java/com/containersol/minimesos/mesos/MesosMaster.java (hunk 4 above)
Comment (frankscholten) [inline minimesos/src/main/java/com/containersol/minimesos/mesos/MesosMaster.java]: Perhaps remove the method at MesosContainer? It is then implemented in MesosMaster or MesosAgent.

File: minimesos/src/main/groovy/com/containersol/minimesos/config/ClusterConfig.groovy [hunk 6]
```diff
@@ -53,4 +56,14 @@ class ClusterConfig extends GroovyBlock {
         delegateTo(marathon, cl)
     }
 
+    void setLoggingLevel(String loggingLevel) {
+        if (!StringUtils.equalsIgnoreCase(loggingLevel, "WARNING") && !StringUtils.equalsIgnoreCase(loggingLevel, "INFO") && !StringUtils.equalsIgnoreCase(loggingLevel, "ERROR")) {
+            throw new RuntimeException("Property 'loggingLevel' can only have the values INFO, WARNING or ERROR")
+        }
+        this.loggingLevel = loggingLevel
+    }
```
Comment (sadovnikov) [inline minimesos/src/main/groovy/com/containersol/minimesos/config/ClusterConfig.groovy]: This one is very different now from implementation in `MesosContainerConfig`
```
+    public void setLoggingLevel(String loggingLevel) {
 +        this.loggingLevel = loggingLevel.toUpperCase()
 +    }
```

`WARN`, not `WARNING` is commonly used logging level. Should we use it?

**Pull request merged**